cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/dataset/MultipleOutputs.java

```diff
@@ -0,0 +1,280 @@
+  }
+
+  // Returns the named output OutputFormat.
+  static Class<? extends OutputFormat> getNamedOutputFormatClass(JobContext job, String namedOutput) {
+    return job.getConfiguration().getClass(MO_PREFIX + namedOutput + FORMAT, null, OutputFormat.class);
+  }
+
+  // Returns the key class for a named output.
+  private static Class<?> getNamedOutputKeyClass(JobContext job, String namedOutput) {
+    return job.getConfiguration().getClass(MO_PREFIX + namedOutput + KEY, null, Object.class);
+  }
+
+  // Returns the value class for a named output.
+  private static Class<?> getNamedOutputValueClass(JobContext job, String namedOutput) {
+    return job.getConfiguration().getClass(MO_PREFIX + namedOutput + VALUE, null, Object.class);
+  }
+
+  static Map<String, String> getNamedConfigurations(JobContext job, String namedOutput) {
+    Map<String, String> namedConf = new HashMap<>();
+
+    String confKeyPrefix = MO_PREFIX + namedOutput + CONF;
+    Map<String, String> properties = job.getConfiguration().getValByRegex(confKeyPrefix + ".*");
+    for (Map.Entry<String, String> entry : properties.entrySet()) {
+      namedConf.put(entry.getKey().substring(confKeyPrefix.length()), entry.getValue());
+    }
+    return namedConf;
+  }
+
+  @VisibleForTesting
+  static void setNamedConfigurations(Job job, String namedOutput, Map<String, String> namedConf) {
+    String confKeyPrefix = MO_PREFIX + namedOutput + CONF;
+    for (Map.Entry<String, String> entry : namedConf.entrySet()) {
+      job.getConfiguration().set(confKeyPrefix + entry.getKey(), entry.getValue());
+    }
+  }
+
+  /**
+   * Adds a named output for the job.
+   *
+   * @param job               job to add the named output
+   * @param namedOutput       named output name, it has to be a word, letters
+   *                          and numbers only, cannot be the word 'part' as
+   *                          that is reserved for the default output.
+   * @param outputFormatClass name of the OutputFormat class.
+   * @param keyClass          key class
+   * @param valueClass        value class
+   * @param outputConfigs     configurations for the output
+   */
+  @SuppressWarnings("unchecked")
+  public static void addNamedOutput(Job job, String namedOutput, String outputFormatClass,
+                                    Class<?> keyClass, Class<?> valueClass, Map<String, String> outputConfigs) {
+    checkNamedOutputName(namedOutput, getNamedOutputsList(job), false);
+    Configuration conf = job.getConfiguration();
+    conf.set(MULTIPLE_OUTPUTS, conf.get(MULTIPLE_OUTPUTS, "") + " " + namedOutput);
+    conf.set(MO_PREFIX + namedOutput + FORMAT, outputFormatClass);
+    conf.setClass(MO_PREFIX + namedOutput + KEY, keyClass, Object.class);
+    conf.setClass(MO_PREFIX + namedOutput + VALUE, valueClass, Object.class);
+    setNamedConfigurations(job, namedOutput, outputConfigs);
+  }
+
+  // instance code, to be used from Mapper/Reducer code
+
+  private TaskInputOutputContext context;
```

Comment:
Fields should be define on the top. Don't adapt the Hadoop coding style. It's horrible.